recipes/boost/all/test_package/conanfile.py

```diff
@@ -59,4 +59,4 @@ def test(self):
         if not can_run(self):
             return
         with chdir(self, self.folders.build_folder):
-            self.run(f"ctest --output-on-failure -C {self.settings.build_type}", env="conanrun")
+            self.run(f"ctest --output-on-failure -C {self.settings.build_type}")
```

Comment:
env="conanrun" must not be removed

Reply:
I see, but I think `test()` @memsharded is referring on (`CMake.test()`?) is totally different than `test()` of test package and `self.run()` command we are running.
So I don't say there is not also an issue here when someone has `cmake` in `tool_requires`, but by default a raw `self.run` is executed in build scope, so I don't see how it couldn't lead to a regression. What you want is to inject env vars of cmake into run scope, or inject both build & run scope into this self.run

Reply:
Hi @Spacelm,
I agree the circumstances in the thread with @memsharded were slightly different, but they are, in fact, a subset of what we have here. In both instances, the test_package test() method is calling ctest, so we need to be able to find that, which may be provided as part of a Conan-managed toolchain via [tool_requires]. The second issue, and it perhaps suggests that we really do need a [test_requires] feature in the host profile, is that the tests that use dynamic linking need to be able to see the runtime shared libraries provided by the compiler (GCC in this case) but those appear to not be part of the runtime environment when GCC is a [tool_requires]. I will add this new use case to the Conan issue thread to see what insights @memsharded can provide. Thanks.

Reply:
The removal of the env altogether will cause the tests to fail on Windows when `shared=True`, CI will fail in this instance:
```

The following tests FAILED:
	  1 - boost_random (Exit code 0xc0000135
)
	  3 - boost_test (Exit code 0xc0000135
)
	  4 - coroutine_test (Exit code 0xc0000135
)
	  5 - chrono_test (Exit code 0xc0000135
)
	  6 - boost_fiber (Exit code 0xc0000135
)
	  7 - boost_json (Exit code 0xc0000135
)
	  8 - boost_nowide (Exit code 0xc0000135
)
	  9 - boost_locale (Exit code 0xc0000135
```

where the exit code matches `STATUS_DLL_NOT_FOUND`

Based on what you are describing, I believe the changes introduced in https://github.com/conan-io/conan-center-index/pull/16348 should handle the case where you are expecting the gcc runtime libraries from a `tool_requires`. with an interest of moving forward with this PR, I'll restore this to how it is on `master` currently, and if there are still issues running tests in this instance, we han investigate separately - as it's something that is not specific to the boost test_package.